src/libs/fsdrv/lv_fs_arduino_sd.cpp

```diff
@@ -0,0 +1,187 @@
+#include <SPI.h>
+#include "SD.h"
+
+typedef struct SdFile {
+    File file;
+} SdFile;
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+static void fs_init(void);
+static void * fs_open(lv_fs_drv_t * drv, const char * path, lv_fs_mode_t mode);
+static lv_fs_res_t fs_close(lv_fs_drv_t * drv, void * file_p);
+static lv_fs_res_t fs_read(lv_fs_drv_t * drv, void * file_p, void * buf, uint32_t btr, uint32_t * br);
+static lv_fs_res_t fs_write(lv_fs_drv_t * drv, void * file_p, const void * buf, uint32_t btw, uint32_t * bw);
+static lv_fs_res_t fs_seek(lv_fs_drv_t * drv, void * file_p, uint32_t pos, lv_fs_whence_t whence);
+static lv_fs_res_t fs_tell(lv_fs_drv_t * drv, void * file_p, uint32_t * pos_p);
+
+/**
+ * Register a driver for the SD File System interface
+ */
+extern "C" void lv_fs_arduino_sd_init(void)
+{
+    fs_init();
+
+    lv_fs_drv_t * fs_drv = &(LV_GLOBAL_DEFAULT()->arduino_sd_fs_drv);
+    lv_fs_drv_init(fs_drv);
+
+    fs_drv->letter = LV_FS_ARDUINO_SD_LETTER;
+    fs_drv->open_cb = fs_open;
+    fs_drv->close_cb = fs_close;
+    fs_drv->read_cb = fs_read;
+    fs_drv->write_cb = fs_write;
+    fs_drv->seek_cb = fs_seek;
+    fs_drv->tell_cb = fs_tell;
+
+    fs_drv->dir_close_cb = NULL;
+    fs_drv->dir_open_cb = NULL;
+    fs_drv->dir_read_cb = NULL;
+
+    lv_fs_drv_register(fs_drv);
+}
+
+/**********************
+ *   STATIC FUNCTIONS
+ **********************/
+
+/*Initialize your Storage device and File system.*/
+static void fs_init(void)
+{
+    if(!SD.begin(LV_FS_ARDUINO_SD_CS_PIN, SPI, LV_FS_ARDUINO_SD_FREQUENCY)) {
+        LV_LOG_WARN("Driver Arduino SD Card not mounted");
+        return;
+    }
+
+    LV_LOG_WARN("Driver Arduino SD Card mounted");
+}
+
+/**
+ * Open a file
+ * @param drv       pointer to a driver where this function belongs
+ * @param path      path to the file beginning with the driver letter (e.g. S:/folder/file.txt)
+ * @param mode      read: FS_MODE_RD, write: FS_MODE_WR, both: FS_MODE_RD | FS_MODE_WR
+ * @return          a file descriptor or NULL on error
+ */
+static void * fs_open(lv_fs_drv_t * drv, const char * path, lv_fs_mode_t mode)
+{
+    LV_UNUSED(drv);
+
+    const char * flags;
+    if(mode == LV_FS_MODE_WR)
+        flags = FILE_WRITE;
+    else if(mode == LV_FS_MODE_RD)
+        flags = FILE_READ;
+    else if(mode == (LV_FS_MODE_WR | LV_FS_MODE_RD))
+        flags = FILE_WRITE;
+
+    File file = SD.open(path, flags);
+    if(!file) {
+        return NULL;
+    }
+
+    SdFile * lf = new SdFile{file};
+
+    return (void *)lf;
+}
+
+/**
+ * Close an opened file
+ * @param drv       pointer to a driver where this function belongs
+ * @param file_p    pointer to a file_t variable. (opened with fs_open)
+ * @return          LV_FS_RES_OK: no error or  any error from @lv_fs_res_t enum
+ */
+static lv_fs_res_t fs_close(lv_fs_drv_t * drv, void * file_p)
+{
+    LV_UNUSED(drv);
+    SdFile * lf = (SdFile *)file_p;
+    lf->file.close();
+    lv_free(lf);
```

Comment:
Why not `delete lf`’?

Reply:
As far as I know, regarding the `lv_free` code, it depends on what's used behind it. It calls `lv_free_core` which can be used by `clib`, `micropython`, or `rtthread`.
It has been asked on another PR to prefer to use `lv_free` over `delete` I think for this reason.

Reply:
As the file was allocated with `new` [here](https://github.com/lvgl/lvgl/pull/5968/files#diff-ff738ff520e493610c7223eaebc4ca92c9802377c56e89d0187ce0d1433731caR87) I agree that `delete` shall be used here.

Reply:
Did it in my last squash, really sorry for missing that :)